omegaconf/base.py

```diff
@@ -321,32 +306,76 @@ def _select_impl(
             throw_on_type_error=throw_on_resolution_failure,
         )
         if value is None:
-            return root, last_key, value
+            return root, last_key, None
         value = root._resolve_interpolation(
+            parent=root,
             key=last_key,
             value=value,
             throw_on_missing=False,
             throw_on_resolution_failure=True,
         )
         return root, last_key, value
 
+    def _resolve_complex_interpolation(
```

Comment:
@omry we had a potentially unresolved discussion on the need for splitting `_resolve_interpolation()` in two: https://github.com/omry/omegaconf/pull/321#discussion_r507186814

Please have a look a the previous discussion and the combined diff and let me know if it requires further clarification.

Reply:
Final comments seems reasonable. I may have more questions once I review the whole thing.